File: app/src/main/java/com/petnbu/petnbu/ui/BindingAdapters.java
package com.petnbu.petnbu.ui;

import android.databinding.BindingAdapter;
import android.view.View;
import android.widget.ImageView;

import com.bumptech.glide.Glide;
import com.petnbu.petnbu.model.Photo;
import com.petnbu.petnbu.util.ImageUtils;

public class BindingAdapters {

    @BindingAdapter("imageUrl")
    public static void bindImage(ImageView imageView, String url) {
        Glide.with(imageView).load(url).into(imageView);
    }

    @BindingAdapter("bindPhoto")
    public static void bindImage(ImageView imageView, Photo photo) {
        new ImageUtils.SizeDeterminer(imageView).getSize((width, height) -> {
            String url = ImageUtils.getPhotoUrl(photo, ImageUtils.getResolutionType(width));
            Glide.with(imageView).load(url).into(imageView);
        });
    }

    @BindingAdapter("visibleGone")
    public static void showHide(View view, boolean show) {
        view.setVisibility(show ? View.VISIBLE : View.GONE);
    }
}

File: app/src/main/java/com/petnbu/petnbu/ui/BindingAdapters.kt
package com.petnbu.petnbu.ui

import android.databinding.BindingAdapter
import android.view.View
import android.widget.ImageView

import com.bumptech.glide.Glide
import com.petnbu.petnbu.model.Photo
import com.petnbu.petnbu.util.ImageUtils

object BindingAdapters {

    @JvmStatic
    @BindingAdapter("imageUrl")
    fun bindImage(imageView: ImageView, url: String) {
        Glide.with(imageView).load(url).into(imageView)
    }

    @JvmStatic
    @BindingAdapter("bindPhoto")
    fun bindImage(imageView: ImageView, photo: Photo) {
        ImageUtils.SizeDeterminer(imageView).getSize { width, _ ->
            val url = ImageUtils.getPhotoUrl(photo, ImageUtils.getResolutionType(width))
            Glide.with(imageView).load(url).into(imageView)
        }
    }

    @JvmStatic
    @BindingAdapter("visibleGone")
    fun showHide(view: View, show: Boolean) {
        view.visibility = if (show) View.VISIBLE else View.GONE
    }
}
